bin/generated_template.d.ts

```diff
@@ -0,0 +1,83 @@
+export interface FilePosition {
+  offset: number;
+  line: number;
+  column: number;
+}
+export interface FileRange {
+  start: FilePosition;
+  end: FilePosition;
+  source: string;
+}
+export interface LiteralExpectation {
+  type: "literal";
+  text: string;
+  ignoreCase: boolean;
+}
+export interface ClassParts extends Array<string | ClassParts> {
+}
+export interface ClassExpectation {
+  type: "class";
+  parts: ClassParts;
+  inverted: boolean;
+  ignoreCase: boolean;
+}
+export interface AnyExpectation {
+  type: "any";
+}
+export interface EndExpectation {
+  type: "end";
+}
+export interface OtherExpectation {
+  type: "other";
+  description: string;
+}
+export type Expectation =
+  | AnyExpectation
+  | ClassExpectation
+  | EndExpectation
+  | LiteralExpectation
+  | OtherExpectation;
+
+declare class _PeggySyntaxError extends Error {
+  static buildMessage(expected: Expectation[], found: string | null): string;
+  message: string;
+  expected: Expectation[];
+  found: string | null;
+  location: FileRange;
+  name: string;
+  constructor(
+    message: string,
+    expected: Expectation[],
+    found: string | null,
+    location: FileRange,
+  );
+  format(sources: {
+    source?: any;
```

Comment:
Is there any way I could avoid `any` here?

Reply:
It's either that, `unknown`, or we put in another generic type, which will make the whole thing kind of a mess.

It really can be almost anything, as long as it can be converted to a string with `String(source)`, and it matches the `grammarSource` that got passed in to `parse`.